mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -536,28 +574,121 @@ push_object (GCObject *obj)
 		data = create_data (obj);
 	g_assert (data->state == INITIAL);
 	g_assert (data->index == -1);
+
+	*forward_obj = data;
+	*push_obj = data;
+}
+
+static void
+mark_object_scanned (ScanData *data)
+{
+	data->low_index = data->index = object_index++;
 	dyn_array_ptr_push (&scan_stack, data);
+	dyn_array_ptr_push (&loop_stack, data);
+}
+
+// The target of the scan gets pushed onto the stack first.
+// However, when using OPTIMIZATION_FORWARD, we want to avoid pushing the object
+// if it has exactly one forwardable reference. Therefore we wait to push the
+// scan target until we know this criteria has failed.
```

Comment:
Is scan target the "obj" variable? Then maybe name the variable that and use @scan_target in the comment to make the connection between docs and code more direct.